### ISSUE TITLE
RequestTimeout is not passed to HttpClient since v6.8.7

### ISSUE THREAD

**Issue by aldobrynin** [elastic/elasticsearch-net#6062]
**NEST/Elasticsearch.Net version**:
Elasticsearch.NET 6.8.9 (.NET 5)
**Elasticsearch version**:
reproduced with 6.2.4 and 6.8.14

**Description of the problem including expected versus actual behavior**:
When trying to execute any long-running request there is no way to override `HttpClient`'s default timeout of 100 seconds.

**Steps to reproduce**:
1. Specify RequestTimeout greater than 100 seconds in request parameters:
```
var parameters = new ForceMergeRequestParameters
{
      RequestConfiguration = new RequestConfiguration {RequestTimeout = TimeSpan.FromMinutes(60)}
};
```
 2. Execute some long-running request with these parameters (make sure ES doesn't process request faster than the timeout) :
 ```
client.IndicesForcemerge<StringResponse>(indexName, parameters);
```
 3. After 100 seconds you'll get exception:
 ```
Caused by: System.Threading.Tasks.TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
      at System.Net.Http.HttpClient+<SendAsyncCore>d__85.MoveNext()
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess()
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification()
      at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
      at Elasticsearch.Net.HttpConnection.Request(C:\Users\SteveGordon\Code\elasticsearch-net\src\Elasticsearch.Net\Connection\HttpConnection-CoreFx.cs:61:5)
    Caused by: System.TimeoutException: The operation was canceled.
      Caused by: System.Threading.Tasks.TaskCanceledException: The operation was canceled.
          at System.Net.Http.HttpConnection+<SendAsyncCore>d__56.MoveNext()
          at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
          at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess()
          at System.Net.Http.HttpConnectionPool+<SendWithRetryAsync>d__72.MoveNext()
          at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
          at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess()
          at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification()
          at System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()
          at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
          at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess()
          at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification()
          at System.Net.Http.HttpClient+<SendAsyncCore>d__85.MoveNext()
        Caused by: System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..
            at System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException()
            at System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.GetResult()
            at System.Net.Http.HttpConnection+<FillAsync>d__94.MoveNext()
            at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess()
            at System.Net.Http.HttpConnection+<ReadNextResponseHeaderLineAsync>d__91.MoveNext()
            at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess()
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification()
            at System.Net.Http.HttpConnection+<SendAsyncCore>d__56.MoveNext()
          Caused by: System.Net.Sockets.SocketException: The I/O operation has been aborted because of either a thread exit or an application request.
```

**Expected behavior**
HttpClient uses timeout specified in `RequestConfiguration.RequestTimeout`

This behavior seems to be broken since 6.8.7 (https://github.com/elastic/elasticsearch-net/commit/903bba2236f3fb1555bdf24fe6b7bf72b8ddf652#diff-1bbac8a8fd708132d878b2b0e44ab95fc379fd0d2515e59e7264c5fecabfd061R87)

**Comment by aldobrynin** [comment]
Seems like this was already fixed for 7.x in https://github.com/elastic/elasticsearch-net/pull/4846, is it possible to backport it to the latest 6.8.x?

**Comment by stevejgordon** [comment]
Thanks for catching this @aldobrynin and for opening the PR. That's merged now. I'll keep this open until I have a patch release scheduled to include the fix.

**Comment by aldobrynin** [comment]
Thanks a lot! is there any specific date when you plan to release this patch?

**Comment by stevejgordon** [comment]
@aldobrynin I've released 6.8.10 to Nuget today so this should soon be available to pull. Thanks again!
